runtime/caches/mod.ts

```diff
@@ -1,20 +1,53 @@
 import { withInstrumentation } from "./common.ts";
-import { compose } from "./compose.ts";
 import { caches as cachesKV } from "./denoKV.ts";
 import { caches as cachesProxy } from "./proxy.ts";
 import { caches as redisCache, redis } from "./redis.ts";
+import { caches as cachesFileSystem } from "./fileSystem.ts";
+import { caches as cachesS3 } from "./s3.ts";
+import { createTieredCache } from "./tiered.ts";
 
-const DEFAULT_CACHE_ENGINE: CacheEngine = "CF_PROXY";
+const ENABLE_LOADER_CACHE = Deno.env.get("ENABLE_LOADER_CACHE") === "true";
+const DEFAULT_CACHE_ENGINE = ENABLE_LOADER_CACHE ? "KV" : "CF_PROXY";
 const WEB_CACHE_ENGINES: CacheEngine[] = Deno.env.has("WEB_CACHE_ENGINE")
   ? Deno.env.get("WEB_CACHE_ENGINE")!.split(",") as CacheEngine[]
-  : [DEFAULT_CACHE_ENGINE];
+  : DEFAULT_CACHE_ENGINE.split(",") as CacheEngine[];
+
+const FILE_SYSTEM_CACHE_DIRECTORY =
+  Deno.env.get("FILE_SYSTEM_CACHE_DIRECTORY") ?? undefined;
+
+const bucketName = Deno.env.get("CACHE_UPLOAD_BUCKET");
+const awsRegion = Deno.env.get("CACHE_AWS_REGION");
+const awsAccessKeyId = Deno.env.get("CACHE_AWS_ACCESS_KEY_ID")!;
+const awsSecretAccessKey = Deno.env.get("CACHE_AWS_SECRET_ACCESS_KEY")!;
+const awsEndpoint = Deno.env.get("CACHE_AWS_ENDPOINT");
```

Comment:
this is duplicated from s3.ts file, you should either:


1) pass as a parameter when creating the s3 caches
2) put back inside the s3.ts file (sorry I think it might be an option to put back the previous approach I think its really better)